### PR TITLE
Add metrics to indicate flow dispatch failure/success count

### PR DESCRIPTION
**Pull request by jamiesjc**

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java [hunk 1]
```diff
@@ -34,6 +34,8 @@
   private final MetricsManager metricsManager;
   private Meter dbConnectionMeter;
   private Meter flowFailMeter;
+  private Meter dispatchFailMeter;
```
Comment (suvodeep-pyne) [inline azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java]: I think we should have a better way of organizing metrics. Keeping all metrics together in one class seems like putting apples and oranges in one basket. Also, such organization of metrics hints that metrics events can be declared from anywhere else in the code. However, dispatch fail / success should only be controlled by the `ExecutorManager`. Thoughts?

Comment (kunkun-tang) [inline azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java]: Sounds a not bad idea, but it is difficult to do at today's codebase, since Executor still exists in common module.

Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java]: I agree with @suvodeep-pyne.

File: azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java [hunk 2]
```diff
@@ -34,6 +34,8 @@
...
+  private Meter dispatchSuccessMeter;
```
Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java]: Aren't these metrics web server only? If so, should they be in azkaban.webapp.WebMetrics?

Comment (jamiesjc) [inline azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java]: These metrics are for executorManager, and executorManager is in azkaban-common. Unless we refactor executorManager and put it in web server, I cannot think of a way to do that properly.

File: azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java (hunk 1 above)
Comment (jamiesjc) [inline azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java]: I agree. All the metrics should be refactored in a separate pull request.

File: azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java (hunk 2 above)
Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java]: I see. Thanks

File: azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java (hunk 1 above)
Comment (suvodeep-pyne) [inline azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java]: Oh cool. @jamiesjc, In that case, should we just retain these Meters in `ExecutorManager` then instead? The existing metrics can be refactored separately in a different PR.
